geth/common/queued_tx.go

```diff
@@ -0,0 +1,144 @@
+package common
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// QueuedTx holds enough information to complete the queued transaction.
+type QueuedTx struct {
+	id         QueuedTxID
+	hash       common.Hash
+	context    context.Context
```

Comment:
Context is normally `ctx`. Also would add mutex as field `mu`. Otherwise methods are public and could be directly used by external code.